### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by thinhbd-1256**
Implement Advanced login

### CODE REVIEW

**Comment by thinhbd-1256** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -174,3 +174,17 @@ input {
     color: $state-danger-text;
   }
 }
+
+.checkbox {
+  margin-top: -10px;
+  margin-bottom: 10px;
```
Comment (huongnguyenmta) [inline app/assets/stylesheets/custom.scss]: gộp 179. 180

**Comment by huongnguyenmta** [comment]
commented

**Comment by thinhbd-1256** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -174,3 +173,16 @@ input {
     color: $state-danger-text;
   }
 }
+
+.checkbox {
+  margin-top: -10px 0 10px 0;
```
Comment (hungnh103) [inline app/assets/stylesheets/custom.scss]: `margin-top: -10px 0 10px;`

File: app/models/user.rb [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
```
Comment (hungnh103) [inline app/models/user.rb]: em dùng method `update_attributes` thay cho `update_attribute` nhé

**Comment by hungnh103** [comment]
commented
Em fix xong cmt rồi merge nhé

**Pull request merged**